server/src/test/java/org/elasticsearch/index/reindex/BulkByScrollResponseTests.java

```diff
@@ -137,19 +143,26 @@ protected void assertEqualInstances(BulkByScrollResponse expected, BulkByScrollR
 
     @Override
     protected BulkByScrollResponse createTestInstance() {
-        // failures are tested separately, so we can test XContent equivalence at least when we have no failures
-        return
-            new BulkByScrollResponse(
-                timeValueMillis(randomNonNegativeLong()), BulkByScrollTaskStatusTests.randomStatusWithoutException(),
-                emptyList(), emptyList(), randomBoolean()
-            );
+        if (testExceptions) {
+            return new BulkByScrollResponse(timeValueMillis(randomNonNegativeLong()), BulkByScrollTaskStatusTests.randomStatus(),
+                randomIndexingFailures(), randomSearchFailures(), randomBoolean());
+        } else {
+            return new BulkByScrollResponse(timeValueMillis(randomNonNegativeLong()),
+                BulkByScrollTaskStatusTests.randomStatusWithoutException(), emptyList(), emptyList(), randomBoolean());
+        }
     }
 
     @Override
     protected BulkByScrollResponse doParseInstance(XContentParser parser) throws IOException {
         return BulkByScrollResponse.fromXContent(parser);
     }
 
+    @Override
+    protected boolean assertToXContentEquivalence() {
+        // XContentEquivalence fails in the exception case, due to how exceptions are serialized.
```

Comment:
I think this means that
toXContext(x) != toXContent(fromXContent(toXContent(x)))
such that we loose more info on the second toXContent? I would think that we would like to keep this check, is there a good reason why we cannot do this?

Reply:
Yes serializing exceptions multiple times is not equivalent. You serialize resource not found exception, and you lose the resource not found on the deserialization. So then the next time you serialize it, it is represented differently. 

I should point out I have maintained this check in the success case (which is the only case tested prior to this PR).